File: source/samples/Sha265Sign.java
import java.io.UnsupportedEncodingException;
import java.net.URI;
import java.net.URISyntaxException;
import java.net.URLEncoder;
import java.security.InvalidKeyException;
import java.security.NoSuchAlgorithmException;
import java.security.SignatureException;
import java.util.HashMap;
import java.util.Iterator;
import java.util.Map;
import java.util.Map.Entry;
import java.util.TreeMap;
import javax.crypto.Mac;
import javax.crypto.spec.SecretKeySpec;
import org.apache.commons.codec.binary.Base64;



class Main {
  private static final String CHARACTER_ENCODING = "UTF-8";
  final static String ALGORITHM = "HmacSHA256";

  public static void main(String[] args) throws Exception {
    String payload="<body JSON string>";
    String action="<action>";
    String url="<url>";
    String timestamp="<timestamp>";
    String data=new String(Base64.encodeBase64((action+url+timestamp+payload).getBytes(CHARACTER_ENCODING)));
    System.out.println();
    String base64Key="<signing key>";
    String signature = sign(data,base64Key);
    System.out.println(signature);
  }

  private static String sign(String data, String secretKey)
      throws NoSuchAlgorithmException, InvalidKeyException, IllegalStateException, UnsupportedEncodingException {
    Mac mac = Mac.getInstance(ALGORITHM);
    mac.init(new SecretKeySpec(Base64.decodeBase64(secretKey), ALGORITHM));
    byte[] signature = mac.doFinal(Base64.decodeBase64(data));
    String signatureBase64 = new String(Base64.encodeBase64(signature), CHARACTER_ENCODING);
    return new String(signatureBase64);
  }
}

File: source/samples/Sha265Sign.js
const crypto = require('crypto');

function sign(data, secretKey) {
  const key = Buffer.from(secretKey, 'base64');
  const hmac = crypto.createHmac('sha256', key);
  const dataBuffer = Buffer.from(data, 'base64');
  hmac.update(dataBuffer);
  return hmac.digest('base64');
}

async function main() {
  const payload = "<body JSON string>";  // Replace with actual payload
  const action = "<action>";            // Replace with actual action
  const url = "<url>";                  // Replace with actual URL
  const timestamp = "<timestamp>";      // Replace with actual timestamp

  const dataString = action + url + timestamp + payload;
  const encodedData = Buffer.from(dataString).toString('base64');

  console.log("Encoded Data:", encodedData);

  const base64Key = "<signing key>";    // Replace with actual base64Key
  const signature = sign(encodedData, base64Key);

  console.log("Signature:", signature);
}

main().catch(console.error);